server/base/src/main/java/org/apache/accumulo/server/security/SystemCredentials.java

```diff
@@ -83,10 +82,19 @@ public TCredentials toThrift(String instanceID) {
   public static final class SystemToken extends PasswordToken {
 
     /**
-     * Accumulo servers will only communicate with each other when this is the same. Bumped for 2.0
-     * to prevent 1.9 and 2.0 servers from communicating.
+     * Accumulo servers will only communicate with each other when this is the same.
+     *
+     * <ul>
+     * <li>Initial version 2 for 1.5 to support rolling upgrades for bugfix releases (ACCUMULO-751)
+     * <li>Bumped to 3 for 1.6 for additional namespace RPC changes (ACCUMULO-802)
+     * <li>Bumped to 4 for 2.0 to prevent 1.9 and 2.0 servers from communicating (#1139)
+     * <li>Bumped to 5 for 2.1 because of system credential hash incompatibility (#1798 / #1810)
+     * </ul>
      */
-    private static final Integer INTERNAL_WIRE_VERSION = 4;
+    static final int INTERNAL_WIRE_VERSION = 5;
+
+    static final String SALT_PREFIX = "$6$"; // SHA-512
```

Comment:
It looks like this is required by the Crypt library?  Would be nice to mention where these values for the salt string are coming from.

Reply:
Actually couldn't we use their constant in `Sha2Crypt` ?
https://commons.apache.org/proper/commons-codec/jacoco/org.apache.commons.codec.digest/Sha2Crypt.java.html

Reply:
This is the well-known scheme for crypt(3) formatted strings. I think it's self-explanatory, since we're specifically using it with the Crypt API (which has good javadocs) and because crypt(3) is well-known. See the man page at https://www.man7.org/linux/man-pages/man3/crypt.3.html

Reply:
While it is more obvious, once you read 3/4 of the man page, most people don't read entire man pages or won't know that this library is based off of the linux standard.  It is generally good practice to avoid hard coding values when you can.

Reply:
Nevermind, the constant is package private so can't be used outside the package anyway.

Reply:
> Actually couldn't we use their constant in `Sha2Crypt` ?

That's what I had originally wanted to do. :/

> While it is more obvious, once you read 3/4 of the man page,

I should be more clear: I didn't mean it is obvious that the library is based on the crypt(3) POSIX standard, or that it was obvious what the constant was for. Rather, I meant to say that the only comment I could think to add would have been self-explanatory, as my best attempt looked like "see Crypt.crypt()". If there's some more helpful comment to add, I'm okay with adding it... I just couldn't think of anything at the time (or now) that would actually be useful and not self-explanatory.

Reply:
That's fine, the commons Crypt javadoc has enough information.  I would just liked to have used their constant.  Oh well.